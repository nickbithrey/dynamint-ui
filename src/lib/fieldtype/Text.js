import React from 'react';
import PropTypes from 'prop-types';
import { TextField } from 'office-ui-fabric-react/lib/TextField';

const Text = ({name, label, value, update, readOnly, ...remainingProps}) => (
	<TextField 
		label={label} 
		defaultValue={value} 
		onBlur={e => update(e, e.target.value)} 
		{...remainingProps}
	/>
);

Text.defaultProps = {
	value: ''
}

Text.propTypes = {
	name: PropTypes.string.isRequired,
	label: PropTypes.string,
	value: PropTypes.string,
	update: PropTypes.func
};

export default Text;