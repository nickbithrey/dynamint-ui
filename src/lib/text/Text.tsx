import * as React from 'react';
import { Text, ITextProps } from 'office-ui-fabric-react/lib/Text';

export interface IText extends ITextProps {
    children: React.ReactNode;
}

const T = ({children, ...settings}: IText) => (
    <Text {...settings}>
        {children}
    </Text>
);

export default T;