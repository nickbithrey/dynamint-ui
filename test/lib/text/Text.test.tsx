import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Text from 'lib/text/Text';

describe('<Text />', () => {
   
    const props = {
        className: 'classname'
    }
    
    const content = 'this is the content';
    
    let wrapper: ShallowWrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Text {...props}>{content}</Text>);
    })
    
    it('renders valid tag and matches snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});
