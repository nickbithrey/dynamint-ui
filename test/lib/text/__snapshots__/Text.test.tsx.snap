// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders valid tag and matches snapshot 1`] = `
<Text
  className="classname"
>
  this is the content
</Text>
`;
